File: 53 Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sumMax = float('-inf')
        sumCur = 0

        for num in nums:
            sumCur += num
            
            if sumCur > sumMax:
                sumMax = sumCur
            
            if sumCur < 0:
                sumCur = 0
        
        return sumMax

File: 53 Maximum Subarray.js
var maxSubArray = function(nums) {
    
    let sumCur = 0;
    let sumMax = -Infinity;

    for (let i = 0; i<nums.length; i++) {
        sumCur += nums[i];

        if (sumCur > sumMax) sumMax = sumCur;

        if (sumCur < 0) sumCur = 0;
    }

    return sumMax;
};